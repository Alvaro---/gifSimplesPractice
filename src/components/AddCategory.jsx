import { useState } from "react";
import PropTypes from "prop-types";

// export const AddCategory = ({ setCategories }) => {
export const AddCategory = ({ onAddCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onChangeInput = ({ target }) => {
    // console.log(target);
    // console.log(target.value);
    setInputValue(target.value);
  };

  //   const onSubmit = (e) => {
  //     e.preventDefault();
  //     // console.log(e);
  //     if (inputValue.trim().length <= 1) return; // Si es menor que uno se sale de la funcion
  //     setCategories((categories) => [inputValue, ...categories]);
  //     setInputValue("");
  //   };

  const onSubmit = (e) => {
    // console.log("Hola mundo para el test");
    e.preventDefault();
    // console.log(e);
    if (inputValue.trim().length <= 1) return; // Si es menor que uno se sale de la funcion
    onAddCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    // <form onSubmit={(event) => onSubmit(event)}> //Si el mismo argumento se envia, puede quitarse.
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        // onChange={(e) => setInputValue(e.target.value)}

        // onChange={({ target }) => {
        //   //   console.log(target);
        //   console.log(target.value);
        //   setInputValue(target.value);
        // }}

        onChange={onChangeInput}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onAddCategory: PropTypes.func.isRequired,
};
