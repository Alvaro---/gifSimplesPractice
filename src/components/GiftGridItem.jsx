import PropTypes from "prop-types";

export const GiftGridItem = ({ title, imageUrl, id }) => {
  return (
    <div className="card">
      <img src={imageUrl} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GiftGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
