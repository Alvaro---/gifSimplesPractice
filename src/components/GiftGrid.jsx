// api.giphy.com/v1/gifs/search?api_key=ipAQldA03CsZ7rXyTEXB8fjzytqNQgcT&q=valoran
import { useFetchGifts } from "../hooks/useFetchGifts";
import { GiftGridItem } from "./GiftGridItem";

export const GiftGrid = ({ category }) => {
  //Es una buena practica sacar el getGifs ya que no usara memoria cada re render
  //Nunca se deben llamar funciones fuera de una funcion ya que se llamara de nuevo con cada re render.

  const { images, isloading } = useFetchGifts(category);
  //El hook llama la re renderizacion

  console.log(isloading);

  // TODA LA LOGICA DE CARGA PUEDE IR EN UN CUSTOM HOOK
  //   const [images, setImages] = useState([]);

  //   const getImages = async () => {
  //     const newImages = await getGifs(category);
  //     setImages(newImages);
  //   };

  //   //Use effect no puede ser async. Si se puede usar el then. Pero si no se quiere se puee separar en una funcion y llamarla desde el useEffect
  //   useEffect(() => {
  //     getImages();
  //     // getGifs(category).then((newImages) => setImages(newImages));
  //   }, []);

  return (
    <>
      <h3>{category}</h3>

      {/* <LoadingMessage isloading={isloading} /> */}
      {/* Se puede crear ese componente que tenga la condicion por si mismo. */}
      {isloading && <h2>Cargando...</h2>}

      <div className="card-grid">
        {images.map((image) => (
          <GiftGridItem key={image.id} {...image} />
          //El operador ... enviara ada propiedad como property
        ))}
      </div>
    </>
  );
};
