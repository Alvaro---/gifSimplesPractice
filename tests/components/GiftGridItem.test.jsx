import { render, screen } from "@testing-library/react";
import { GiftGridItem } from "../../src/components/GiftGridItem";

describe("Pruebas en GiftGridItem", () => {
  const title = "saitama";
  const url = "https://test.com/";
  test("should match snapshot", () => {
    const { container } = render(<GiftGridItem title={title} imageUrl={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should show image whit url and alt", () => {
    render(<GiftGridItem title={title} imageUrl={url} />);
    // screen.debug(); // Para ver el snapshot
    // console.log(screen.getByRole("img").src);
    // expect(screen.getByRole("img").src).toBe(url);
    // expect(screen.getByRole("img").alt).toBe(title);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("should show title in component", () => {
    render(<GiftGridItem title={title} imageUrl={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
