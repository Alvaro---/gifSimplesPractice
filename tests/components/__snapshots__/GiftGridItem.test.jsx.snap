// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiftGridItem should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://test.com/"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
